src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -860,8 +860,27 @@ private void generate() {
                         push(op.result());
                     }
                     case InvokeOp op -> {
-                        // @@@ var args
-                        processOperands(op);
+                        if (op.isVarArgs()) {
+                            processOperands(op.operands().subList(0, op.operands().size() - op.varArgOperands().size()));
+                            var varArgOperands = op.varArgOperands();
+                            cob.loadConstant(varArgOperands.size());
+                            var compType = ((ArrayType) op.invokeDescriptor().type().parameterTypes().getLast()).componentType();
+                            var typeKind = TypeKind.fromDescriptor(compType.toNominalDescriptor().descriptorString());
+                            if (TypeKind.REFERENCE.equals(typeKind)) {
+                                var cd = ClassDesc.ofDescriptor(compType.toNominalDescriptor().descriptorString());
+                                cob.anewarray(cd);
+                            } else {
+                                cob.newarray(typeKind);
+                            }
+                            for (int j = 0; j < varArgOperands.size(); j++) {
+                                cob.dup();
```

Comment:
Add a comment that this is duplicating the array value on the stack for the next loop iteration *and* subsequent the invoke instruction.